packages/bpk-component-infinite-scroll/readme.md

```diff
@@ -0,0 +1,110 @@
+# bpk-component-infinite-scroll
+
+> Backpack infinite scroll component.
+
+## Installation
+
+```sh
+npm install bpk-component-infinite-scroll --save-dev
+```
+
+## Usage
+```js
+import React from 'react';
+import PropTypes from 'prop-types';
+import BpkInfiniteScroll from 'bpk-component-infinite-scroll';
+
+const SomeList = ({elements}) => (
+  <div id="list">
+    {elements.map(element => (
+      <div key={element} style={{ height: '50px' }}>
+        {element}
+      </div>
+    ))}
+  </div>
+)
+
+const elementsArray = [
+  'element 1',
+  'element 2',
+  'element 3',
+  'element 4',
+  'element 5',
+  'element 6',
+  'element 7',
+  'element 8',
+  'element 9',
+  'element 10',
+]
+
+const InfiniteList = BpkInfiniteScroll(SomeList);
+
+export default () => (
+  <InfiniteList
+    onRequestElements={
+      (index, nElements) =>
+        Promise.resolve(elementsArray.slice(index, index+nElements))
+    }
+  />
+)
+```
+
+## Accompanying functions
+
+DEFAULT_REQUEST_ELEMENTS
+
+`DEFAULT_REQUEST_ELEMENTS` is a function that takes an array of elements
+and returns another function. The returned function has two parameters
+the current index (i) and the number of elements to load (n) and returns a
+Promise that resolves to the sub-array between the indices i and i+n.
+
+```js
+import React from 'react';
+import PropTypes from 'prop-types';
+import BpkInfiniteScroll, { DEFAULT_REQUEST_ELEMENTS } from 'bpk-component-infinite-scroll';
+
+const SomeList = ({elements}) => (
+  <div id="list">
+    {elements.map(element => (
+      <div key={element} style={{ height: '50px' }}>
+        {element}
+      </div>
+    ))}
+  </div>
+)
+
+const elementsArray = [
+  'element 1',
+  'element 2',
+  'element 3',
+  'element 4',
+  'element 5',
+  'element 6',
+  'element 7',
+  'element 8',
+  'element 9',
+  'element 10',
+]
+
+const InfiniteList = BpkInfiniteScroll(SomeList);
+
+export default () => (
+  <InfiniteList
+    onRequestElements={DEFAULT_REQUEST_ELEMENTS(elementsArray)}
+  />
+)
+```
+
+
+## Props
```

Comment:
can you change the order of the properties? we follow this rule - First required props in alphabetical order then optional props in alphabetical.

Reply:
done

Reply:
The order is still not right. It should be:

```
onItemsFetch
elementsPerScroll
loadingComponent
onClickSeeMore
onScroll
onScrollFinished
seeMoreAfter
seeMoreComponent
```

Reply:
You are right 😅

Reply:
done